lib/main.js

```diff
@@ -157,44 +159,59 @@ module.exports = class Router extends EventEmitter {
                 next();
             });
 
-            fileWriteStream.on('file not saved', error => {
-                next(
-                    boom.wrap(error, 400, 'Generated file could not be saved')
-                );
-            });
+            fileWriteStream.on(
+                'file not saved',
+                this.onError(next, 'Generated file could not be saved')
+            );
 
-            fileWriteStream.on('error', error => {
-                next(boom.wrap(error, 400));
-            });
+            fileWriteStream.on('error', this.onError(next));
 
             const reader = new Reader(fileStreams);
 
-            reader.on('error', error => {
-                next(boom.wrap(error, 400));
-            });
+            reader.on('error', this.onError(next));
 
             reader.on('file found', file => {
                 res.locals.included.push(file);
             });
 
-            reader.on('file not found', file => {
-                res.locals.excluded.push(file);
-            });
+            reader.on('file not found', this.onFileNotFound(res));
 
-            reader.on('pipeline empty', () => {
-                next(
-                    boom.badRequest(
-                        'Could not load any of the resources in the payload from storage'
-                    )
-                );
-            });
+            reader.on('pipeline empty', this.onPipelineEmpty(next));
 
             reader.on('pipeline ready', () => {
                 reader.pipe(fileWriteStream);
             });
         };
     }
 
+    onError(next, message) {
+        return error => {
+            next(boom.wrap(error, 400, message));
+        };
+    }
+
+    onFileNotFound(res) {
+        return file => {
+            res.locals.excluded.push(file);
+        };
+    }
+
+    onNotFound(next) {
+        return () => {
+            next(boom.notFound());
+        };
+    }
+
+    onPipelineEmpty(next) {
+        return () => {
+            next(
+                boom.badRequest(
+                    'Could not load any of the resources in the payload from storage'
+                )
+            );
+        };
+    }
+
```

Comment:
these methods were extracted primarily to avoid mock pain and make testing easier. Some reusability was discovered in the process (`onError` used multiple times)